File: BackEnd/SamaniCrm.Infrastructure/Services/Tenant/ProvisioningNotification.cs
using SamaniCrm.Core.Shared.Enums;

namespace SamaniCrm.Infrastructure.Services.TenantService;

public class ProvisioningNotification
{
    public required string TenantSlug { get; set; }
    public required ProvisioningStepStatus Status { get; set; }
    public string? Message { get; set; }
    public required TenantProvisionStepsEnum CurrentStep { get; set; }
    public required DateTime Timestamp { get; set; }
}

File: Frontend/SamaniCrm/src/app/Tenants/provisioning-tenant/provisioning-notification.ts
import { ProvisioningStepStatus } from '@shared/service-proxies/model/provisioning-step-status';
import { TenantProvisionStepsEnum } from '@shared/service-proxies/model/tenant-provision-steps-enum';

export class ProvisioningNotification {
  tenantSlug!: string;
  status!: ProvisioningStepStatus;
  message!: string;
  currentStep!: TenantProvisionStepsEnum;
  timestamp?: string;
}
